server/src/main/java/io/druid/server/coordination/ZkCoordinator.java

```diff
@@ -86,8 +88,10 @@ public void loadLocalCache()
     }
```

Comment:
So, the really confusing thing about this code is that it *does* log before returning.  The call to `addSegments()` is actually blocking and calls the callback in a `finally`...